prov/acmp/src/acmp.c

```diff
@@ -1774,6 +1845,7 @@ acmp_resolve_dest(struct acmp_ep *ep, struct acm_msg *msg, uint64_t id)
 		acm_log(2, "have address, resolving route\n");
 		acm_increment_counter(ACM_CNTR_ADDR_CACHE);
 		atomic_inc(&ep->counters[ACM_CNTR_ADDR_CACHE]);
+		acmp_acquire_sgid(saddr, dest);
```

Comment:
This approach is assigning an SGID to the acmp_dest structure, which it then uses when sending a PR query.  However, the SGID association with the acmp_dest is temporary, and will be replaced the next time a call is made.

Either there needs to be a 1:1 correlation between an acmp_dest and the source address (SGID or SLID), or the source address needs to be passed to the PR query call.  *If* there is a 1:1 correlation, then the source address should be assigned earlier in this flow, maybe when the acmp_dest is allocated and initialized.  If there is NOT a 1:1 relationship here, then the source addresss needs to be passed through and incorporated into the PR as appropriate.